scripts/sixtyfps_weekly.py

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/python3
+
+# LICENSE BEGIN
+# Copyright (c) 2021 Tobias Hunger <tobias.hunger@gmail.com>
+#
+# SPDX-License-Identifier: GPL-3.0-only
+# This file is also available under commercial licensing terms.
+# Please contact info@sixtyfps.io for more information.
+# LICENSE END
+
+import datetime
+from re import sub
+import subprocess
+import typing
+
+
+def parse_message(input: str) -> typing.Tuple[str, str]:
+    subject = ""
+    message = ""
+
+    for l in input.rstrip().split("\n"):
+        if not l.startswith("    "):
+            continue
+        if subject == "":
+            subject = l[4:]
+            continue
+
+        message += f"{l[4:]}\n"
+
+    return (subject, message.strip())
+
+
+def get_git_log(
+    start_date: datetime.datetime, end_date: datetime.datetime
+) -> typing.Sequence[typing.Tuple[str, str, str, str]]:
+    result = subprocess.run(
+        [
+            "/usr/bin/git",
+            "log",
+            f"--since={start_date.isoformat()}",
+            f"--until={end_date.isoformat()}",
+            "--pretty=format:%w(0)Subject: %s%nAuthor: %ae%nHash: %H%nBody:%n%w(80,4,4)%b%n%w(0)END_OF_BODY%n%n",
+        ],
+        capture_output=True,
+        check=True,
+    )
+
+    return _parse_git_log(result.stdout.decode("utf-8"))
+
+
+def _parse_git_log(log_data: str) -> typing.Sequence[typing.Tuple[str, str, str, str]]:
+    result: typing.List[typing.Tuple[str, str, str, str]] = []
+
+    author = ""
+    subject = ""
+    commit = ""
+    message = ""
+
+    for l in log_data.split("\n"):
+        if l == "":
+            message += "\n"
+        elif l.startswith("Subject: "):
+            subject = l[9:].strip()
+        elif l.startswith("Author: "):
+            author = l[8:].strip()
+        elif l.startswith("Hash: "):
+            commit = l[6:].strip()
+        elif l.startswith("Body:"):
+            message = ""
+        elif l.startswith("    "):
+            message = f"{message}{l[4:]}\n"
+        elif l.startswith("END_OF_BODY"):
+            message = message.strip()
+            message_str = message
+            message_str.replace("\n", "\\\n")
+
+            assert commit and author and subject
+
+            result.append((commit, author, subject, message))
+            commit = ""
+            author = ""
+            subject = ""
+            message = ""
+        else:
+            print(f'ERROR: Unknown line "{l}".')
+
+    return result
+
+
+def generate_header(start_date: datetime.datetime, end_date: datetime.datetime) -> str:
+    start_str = f"{start_date.month}.{start_date.day}.{start_date.year}"
```

Comment:
mm.dd.yyyy  id not a valid date format in any country.
We used `from ddth of mmmmm to ddth of mmmmm yyyy`

Reply:
Yes, you are right. It should look a bit better now.